diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -100,15 +100,7 @@
 xmlb_dep = dependency('xmlb', version: '>= 0.3.14',
                       fallback: ['libxmlb', 'libxmlb_dep'],
                       default_options: ['gtkdoc=false', 'introspection=false'])
-if get_option('zstd-support')
-    zstd_dep = dependency('libzstd', required: get_option('zstd-support'))
-else
-    zstd_dep = dependency('', required: false)
-endif
-if get_option('systemd')
-    libsystemd_dep = dependency('libsystemd', required: get_option('systemd'))
-else
-    libsystemd_dep = dependency('', required: false)
-endif
+zstd_dep = dependency('libzstd', required: get_option('zstd-support'))
+libsystemd_dep = dependency('libsystemd', required: get_option('systemd'))

 if get_option ('gir')
